pkg/tfshim/sdk-v2/walk.go

```diff
@@ -0,0 +1,91 @@
+package sdkv2
+
+import (
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/pulumi/pulumi-terraform-bridge/v3/pkg/tfshim/walk"
+	"github.com/pulumi/pulumi/sdk/v3/go/common/util/contract"
+)
+
+func findSchemaContext(topLevel *schema.Resource, path walk.SchemaPath) schemaContext {
```

Comment:
@iwahbe appreciate quick eyes on this; this maybe something that can be expressed more fluently or lifted to the shim.* layer. What I find handy is working around the inability to reference internal types directly with something like this:

```
type attrSchemaContext struct {
	resource *schema.Resource
	name     string
}
```

Which now gives access to this internal machinery via `ctx.resource.CoreConfigSchema().Attributes[a.name].Optional` etc.

Reply:
I'm not sure we want to expose nesting level info to `shim.*`. Nothing jumps out to me.